nixos/modules/services/system/nix-daemon.nix

```diff
@@ -0,0 +1,263 @@
+/*
+  Declares what makes the nix-daemon work on systemd.
+
+  See also
+   - nixos/modules/config/nix.nix: the nix.conf
+   - nixos/modules/config/nix-remote-build.nix: the nix.conf
+*/
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.nix;
+
+  nixPackage = cfg.package.out;
+
+  isNixAtLeast = versionAtLeast (getVersion nixPackage);
+
+  makeNixBuildUser = nr: {
+    name = "nixbld${toString nr}";
+    value = {
+      description = "Nix build user ${toString nr}";
+
+      /*
+        For consistency with the setgid(2), setuid(2), and setgroups(2)
+        calls in `libstore/build.cc', don't add any supplementary group
+        here except "nixbld".
+      */
+      uid = builtins.add config.ids.uids.nixbld nr;
+      isSystemUser = true;
+      group = "nixbld";
+      extraGroups = [ "nixbld" ];
+    };
+  };
+
+  nixbldUsers = listToAttrs (map makeNixBuildUser (range 1 cfg.nrBuildUsers));
+
+in
+
+{
+  imports = [
+    ../../config/nix.nix
+
```

Comment:
isn't this already included in the modules list?

Reply:
It declares that you'd need a global nix configuration when you want to use the daemon. It's not technically necessary, but it is informative and potentially useful if someone wants to cherry pick modules.

Reply:
> It's not technically necessary, but it is informative and potentially useful if someone wants to cherry pick modules.

AFAIK we don't do this anywhere else and removing/picking any module probably draws in all the time a long tail fo dependencies.

Reply:
#242028